wp-modules/editor/js/src/hooks/usePatternData.ts

```diff
@@ -57,18 +59,39 @@ export default function usePatternData( postMeta: PostMeta ) {
 
 	/**
 	 * Alphabetized block pattern categories for the site editor, mapped for react-select.
+	 * Registered and newly added custom categories are included.
+	 * Needed for including new categories before the post is saved.
 	 */
-	const categories = useSelect( ( select: SelectQuery ) => {
-		return sortAlphabetically(
-			select( 'core' )
-				.getBlockPatternCategories()
-				.map( ( category ) => ( {
+	const combinedCategories = sortAlphabetically(
+		[
+			...postMeta.customCategories.reduce(
+				( acc, categoryLabel ) => {
+					const missingCategory =
+						! patternManager.patternCategories.some(
+							( queriedCategory ) =>
+								queriedCategory.label === categoryLabel
+						);
+
+					return missingCategory
+						? [
+								...acc,
+								{
+									label: categoryLabel,
+									value: convertToSlug( categoryLabel ),
+									pm_custom: true,
+								},
+						  ]
+						: acc;
+				},
```

Comment:
@kienstra since you are reviewing — looking at this again, I think the top part of the reducer should be extracted into its own utility.

It's a bit complex for being untested.

Reply:
OK, sounds good

Reply:
This was covered in [bdcce92](https://github.com/studiopress/pattern-manager/pull/123/commits/bdcce9254f0c1f627f6fe4a495992b74219c0331).